Sources/Sentry/SentryInstallation.m

```diff
@@ -5,38 +5,40 @@
 
 @implementation SentryInstallation
 
-static NSString *volatile installationString;
++ (NSMutableDictionary<NSString*, NSString*>*)installationStringsByCacheDirectoryPaths
+{
+    static dispatch_once_t once;
+    static NSMutableDictionary* dictionary;
+    
+    dispatch_once(&once, ^{
+        dictionary = [NSMutableDictionary dictionary];
+    });
+    return dictionary;
+}
 
-+ (NSString *)id
++ (NSString *)idWithCacheDirectoryPath:(NSString *)cacheDirectoryPath
 {
-    if (nil != installationString) {
-        return installationString;
-    }
     @synchronized(self) {
+        NSString* installationString = self.installationStringsByCacheDirectoryPaths[cacheDirectoryPath];
+        
         if (nil != installationString) {
             return installationString;
         }
-        NSString *cachePath
-            = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES)
-                  .firstObject;
-
+        
+        NSString *cachePath = cacheDirectoryPath;
         NSString *installationFilePath = [cachePath stringByAppendingPathComponent:@"INSTALLATION"];
-
         NSData *installationData = [NSData dataWithContentsOfFile:installationFilePath];
 
         if (nil == installationData) {
             installationString = [NSUUID UUID].UUIDString;
-            NSData *installationStringData =
-                [installationString dataUsingEncoding:NSUTF8StringEncoding];
+            NSData *installationStringData = [installationString dataUsingEncoding:NSUTF8StringEncoding];
             NSFileManager *fileManager = [NSFileManager defaultManager];
-            [fileManager createFileAtPath:installationFilePath
-                                 contents:installationStringData
-                               attributes:nil];
+            [fileManager createFileAtPath:installationFilePath contents:installationStringData attributes:nil];
```

Comment:
`m`: I think we should now check the return value, and if it is `NO`, meaning that we couldn't write to the path, we should log an error with `SENTRY_LOG_ERROR` to at least inform our users with a log statement that something is wrong. I'm sorry; I could have mentioned this already in a previous review.